deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -126,17 +131,18 @@ class PropertySegmentsPerformance(WeakSegmentsAbstractText):
     ignore_properties : Union[Hashable, List[Hashable]] , default: None
         Properties to ignore, if none given checks based on properties variable
     n_top_properties : Optional[int] , default: 10
-        Number of properties to use for segment search. Top properties are selected based on feature importance.
+        Number of properties to use for segment search. Selected at random.
     segment_minimum_size_ratio: float , default: 0.05
         Minimum size ratio for segments. Will only search for segments of
         size >= segment_minimum_size_ratio * data_size.
     alternative_scorer : Tuple[str, Union[str, Callable]] , default: None
         Scorer to use as performance measure, either function or sklearn scorer name.
         If None, a default scorer (per the model type) will be used.
-    loss_per_sample: Union[np.array, pd.Series, None], default: None
-        Loss per sample used to detect relevant weak segments. If pd.Series the indexes should be similar to those in
-        the dataset object provide, if np.array the order should be based on the index order of the dataset object and
-        if None the check calculates loss per sample by via log loss for classification and MSE for regression.
+    score_per_sample: Optional[np.array, pd.Series, None], default: None
+        Score per sample are required to detect relevant weak segments. Should follow the convention that a sample with
+        a higher score mean better model performance on that sample. If provided, the check will also use provided
```

Comment:
"also"? Exclusively no?

Reply:
no, it will use it both for the tree building process and also for the leaf ranking